packages/components/src/input-control/README.md

```diff
@@ -0,0 +1,100 @@
+# InputControl
+
+InputControl components let users enter and edit text. This is an experimental component intended to (in time) merge with or replace [TextControl](../text-control).
+
+## Usage
+
+Render a user interface to input the name of an additional css class.
+
+```js
+import { __experimentalInputControl as InputControl } from '@wordpress/components';
+import { useState } from '@wordpress/compose';
+
+const Example = () => {
+	const [ value, setValue ] = useState( '' );
+
+	return (
+		<InputControl
+			value={ value }
+			onChange={ ( nextValue ) => setValue( nextValue ) }
+		/>
+	);
+};
+```
+
+## Props
+
+### disabled
+
+If true, the `input` will be disabled.
+
+-   Type: `Boolean`
+-   Required: No
+-   Default: `false`
+
+### isPressEnterToChange
+
+If true, the `ENTER` key press is required in order to trigger an `onChange`. If enabled, a change is also triggered when tabbing away (`onBlur`).
+
+-   Type: `Boolean`
+-   Required: No
+-   Default: `false`
+
+### isFloatingLabel
+
+If true, the `label` will render with a floating interaction.
```

Comment:
This description seems a bit vague; even I'm not entirely sure what it means. I assume it means the label appears inside the input until the input is focused? Also, is this prop even a good idea, accessibility-wise?

Also, all this type info belongs in a JSDoc comment, doesn't it?

Reply:
Thanks for noticing this! I'm not sure how best to describe this interaction/design.

It works like the TextFields from Material UI:
https://material-ui.com/components/text-fields/

From a design perspective, it offers more versatility, especially if controls need to be combined in cramped spaces.

As for as accessibility goes, how its setup, I believe it it acceptable from both a motion perspective and aria label + HTML markup perspective.

cc'ing some a11y folks!

@enriquesanchez / @diegohaz  Are there any concerns for having floating labels in Inputs like these?

<img width="241" alt="Screen Shot 2020-05-14 at 3 07 28 PM" src="https://user-images.githubusercontent.com/2322354/81975175-a7eb5c80-95f4-11ea-8820-de32a2f194cd.png">


https://material-ui.com/components/text-fields/

Thank you!

Reply:
I don't have strong opinions about that. But there's this article with arguments against floating labels: [Floating labels are problematic](https://medium.com/simple-human/floating-labels-are-a-bad-idea-82edb64220f6)

And this other one is a response to that: [Are float labels really that problematic after all?](http://mds.is/still-floating-labels/)

Reply:
I always check Adam Silver's articles when in doubt about forms and a11y. That first link Diego shared is from him :)

- Sorry if this is obvious, but where is this floating label going to be used? Only on this control or on all input fields?
- I don't see the floating label in the gif example above, but how would it look like when the floating label is longer than the field itself?

Reply:
@diegohaz + @enriquesanchez Thanks for replying!

> where is this floating label going to be used? Only on this control or on all input fields?

It's situational! By default, floating labels are **disabled**. They're useful if the input is narrower or used in compact spaces. For example, within a smaller Popover.

> but how would it look like when the floating label is longer than the field itself?

Ah! the GIF above does not have a floating label. The best example of this interaction would be from the Material TextField demo:

https://material-ui.com/components/text-fields/

My screen capture -> GIF software doesn't do a good job at capturing the various stages.

The key frames work like this ✨ 

No value, label works like a placeholder:

<img width="219" alt="Screen Shot 2020-05-19 at 12 45 31 PM" src="https://user-images.githubusercontent.com/2322354/82354318-a8a63900-99ce-11ea-900c-3bc58defa4d0.png">

No value, focused, the label floats up:

<img width="235" alt="Screen Shot 2020-05-19 at 12 46 20 PM" src="https://user-images.githubusercontent.com/2322354/82354419-c5427100-99ce-11ea-8da9-113d02840dfe.png">

With value, the label stays up:
<img width="227" alt="Screen Shot 2020-05-19 at 12 46 27 PM" src="https://user-images.githubusercontent.com/2322354/82354451-d0959c80-99ce-11ea-8b52-2d35eaa752f7.png">

Reply:
That doesn't really answer the question, though: what happens when the field label is longer than the field itself?

Reply:
Ah! Gotcha. I misread. My apologies. There isn't handling of this at the moment.

The Material TextField doesn't handle this use case too well either:
https://codesandbox.io/s/material-demo-e49dz

I can try some truncation techniques really quickly. I don't think this should be a blocker though.

Reply:
The big issue is that input labels have to be translated, and some languages will use much longer words than others. An input with a floating label of "left" might be correctly sized, but once you translate that to another language, it may suddenly be twice as long.

WordPress generally prefers decisions over options. If there's not a really good reason to provide an alternate input style, we shouldn't provide it. I think it's okay to merge this PR since the `InputControl` will be experimental, but I don't think we should change the component to stable until we know whether the floating label options should be kept or not.

Reply:
![Screen Capture on 2020-05-19 at 13-52-02](https://user-images.githubusercontent.com/2322354/82361425-d2fcf400-99d8-11ea-9c53-4da6ffafdcd6.gif)

This is the best I could do given the (surprisingly) limited CSS rendering control over `<fieldset>`.

> I don't think we should change the component to stable until we know whether the floating label options should be kept or not

Yup! I agree. We can also drop the floating label feature if it's causing issues.

Reply:
Thanks @ItsJonQ! I was trying to make a point with my question, apologies if it wasn't clear 😃.

I feel that at this point there's some uncertainty re: floating labels (a11y, localization, placement, etc.) that my recommendation is to stick to regular ol' fashioned text labels.

Reply:
> I feel that at this point there's some uncertainty re: floating labels (a11y, localization, placement, etc.) that my recommendation is to stick to regular ol' fashioned text labels.

I agree! Ol' fashion text labels are **default** 😊 

The floating labels are a new feature, which can be experimentally trialed in certain scenarios